modules/core/log/modules/quantity/src/EventSubscriber/LogQuantityEventSubscriber.php

```diff
@@ -91,4 +96,41 @@ public function logDelete(LogEvent $event) {
     }
   }
 
+  /**
+   * Perform actions on quantity delete.
+   *
+   * @param \Drupal\quantity\Event\QuantityEvent $event
+   *   The quantity event.
+   */
+  public function quantityDelete(QuantityEvent $event) {
+
+    // Get the quantity entity from the event.
+    $quantity = $event->quantity;
+
+    // Look up logs that reference the quantity.
+    $log_storage = $this->entityTypeManager->getStorage('log');
+    $query = $log_storage->getQuery();
+    $query->condition('quantity.target_id', $quantity->id());
+    $query->accessCheck(FALSE);
+    $log_ids = $query->execute();
+    /** @var \Drupal\log\Entity\LogInterface[] $logs */
+    $logs = [];
+    if (!empty($log_ids)) {
+      $logs = $log_storage->loadMultiple($log_ids);
+    }
+
+    // Remove references to the quantity from the log and save a revision.
+    foreach ($logs as $log) {
+      $log->set('quantity', array_filter($log->get('quantity')->getValue(), function ($value) use ($quantity) {
+        if (!empty($value['target_id']) && $value['target_id'] == $quantity->id()) {
+          return FALSE;
+        }
+        return TRUE;
+      }));
+      $log->setNewRevision(TRUE);
+      $log->setRevisionLogMessage($this->t('Removed reference to deleted quantity %uuid.', ['%uuid' => $quantity->uuid()]));
+      $log->save();
```

Comment:
Just was thinking it would be nice if there weren't multiple revision messages if you delete multiple quantities at the same time. But that might be hard to implement since this event subscriber reacts to a single quantity being deleted, not multiple.

Reply:
Yea I had the same thought. I don't see any way around it, but I think it's good to have a revision log message.